proposals/lip-0052.md

```diff
@@ -362,712 +379,1832 @@ The module provides the following commands to modify the NFT store.
 
 Transactions executing this command have:
 
-* `moduleID = NFT_MODULE_ID`
-* `commandID  = COMMAND_ID_TRANSFER`
+- `module = MODULE_NAME_NFT`
+- `command = COMMAND_NAME_TRANSFER`
 
 ##### Parameters Schema
 
-The `params` property of an NFT transfer transaction follows the schema `NFTTransferParams`.
+The `params` property of a NFT transfer transaction follows the schema `NFTTransferParamsSchema`.
 
 ```java
-NFTTransferParams = {
+NFTTransferParamsSchema = {
     "type": "object",
     "required": [
         "nftID",
-        "recipientAddress"
+        "recipientAddress",
+        "data"
     ],
     "properties": {
         "nftID": {
-            "type": object,
+            "dataType": "bytes",
+            "length": NFT_ID_LENGTH,
             "fieldNumber": 1,
-            "required": [
-                "chainID",
-                "collection",
-                "index"
-            ],
-            "properties": {
-                "chainID": {
-                    "dataType": "uint32",
-                    "fieldNumber": 1
-                },
-                "collection": {
-                    "dataType": "uint32",
-                    "fieldNumber": 2
-                },
-                "index": {
-                    "dataType": "uint64",
-                    "fieldNumber": 3
-                }
-            }
-        },    
+        },
         "recipientAddress": {
             "dataType": "bytes",
+            "length": ADDRESS_LENGTH,
             "fieldNumber": 2
+        },
+         "data": {
+            "dataType": "string",
+            "maxLength": MAX_DATA_LENGTH,
+            "fieldNumber": 3
         }
     }
 }
 ```
 
-##### Parameters Validity
+##### Verification
+
+```python
+def verify(trs: Transaction) -> None:
 
-The `params` property of an NFT transfer transaction is valid if:
+    senderAddress = sha256(trs.senderPublicKey)[:ADDRESS_LENGTH]
+    nftID = trs.params.nftID
 
-* `recipientAddress` must be a byte array of length `ADDRESS_LENGTH`.
+    if NFTStore(nftID) does not exist:
+        raise Exception("NFT store entry does not exist.")
 
-##### Execution
+    if getNFTOwner(nftID) == OWN_CHAIN_ID:
```

Comment:
This check seems superfluous given the following one. I would remove it.

Reply:
Also, I think the check is wrong, shouldn't we check that the length of the owner matches a chain ID length (so it could be any chain)?